### PR TITLE
SDK-497 [Node.js SDK] Remove Environment variables support

### PR DESCRIPTION
**Pull request by igor-alekseev-dev**
Pull Request Checklist
======================

Please read this for our PR Culture:
------------------------------------
https://incountry.atlassian.net/wiki/spaces/ED/pages/46858869/Pull+requests+culture+Draft

Work relates to:
----------------
Link(s) to Jira Task/User Story/Bug/Epic:
- [SDK-497](https://incountry.atlassian.net/browse/SDK-497)

Code Quality (Before submitting PR)
------------
- [x] Ensure code compiles and passes Sonar Lint.
- [x] Do your best to ensure integration tests pass. If they don't then...
- [x] Remove, ignore, or mark pending any brittle tests.
- [x] Remove commented out code.

Target Release
--------------
- [ ] Please put the target release vehicle this needs to be bundled with.
- [ ] Did you increment the version in package.json? What is the new version?

### CODE REVIEW

File: tests/unit/storage/constructor-tests.ts [hunk 1]
```diff
@@ -13,91 +14,39 @@ chai.use(sinonChai);
...
                 .to.be.rejectedWith(StorageConfigValidationError, `Storage.constructor() call: Validation Error: <StorageOptions>.endpoint should be string but got ${options.endpoint}`);
             }));
           });
 
           it('should not throw error if endpoint missing', async () => {
-            process.env.INC_CLIENT_ID = 'clientId';
-            process.env.INC_CLIENT_SECRET = 'clientSecret';
-            process.env.INC_ENVIRONMENT_ID = 'envId';
-            await expect(createStorage({ encrypt: false })).not.to.be.rejected;
+            await expect(createStorage({ environmentId: '', encrypt: false, oauth: { token: '' } })).not.to.be.rejected;
```
Comment (artemgolozubov) [inline tests/unit/storage/constructor-tests.ts:30]:
```suggestion
            await expect(createStorage({ environmentId: 'envId', encrypt: false, oauth: { clientId: 'clientId', clientSecret: 'clientSecret' } })).not.to.be.rejected;
```

File: tests/unit/storage/constructor-tests.ts [hunk 2]
```diff
@@ -106,53 +55,25 @@ describe('Storage', () => {
...
-          });
-
           describe('clientId', () => {
-            it('should be provided via either options or environment variable', async () => {
+            it('should be provided via options', async () => {
               await Promise.all([{ ...baseOptions, oauth: {} }, { ...baseOptions, oauth: { clientId: undefined } }].map(async (options) => {
+                // @ts-ignore
                 await expect(createStorage(options))
-                  .to.be.rejectedWith(StorageConfigValidationError, 'Please pass clientId in options or set INC_CLIENT_ID env var');
+                  .to.be.rejectedWith(StorageConfigValidationError, 'Storage.constructor() call: Validation Error: <StorageOptions>.oauth.clientSecret should be string but got undefined');
```
Comment (artemgolozubov) [inline tests/unit/storage/constructor-tests.ts:63]:
```suggestion
                  .to.be.rejectedWith(StorageConfigValidationError, 'Storage.constructor() call: Validation Error: <StorageOptions>.oauth.clientId should be string but got undefined');
```

File: tests/unit/validation/storage-options-test.ts [hunk 3]
```diff
@@ -1,139 +1,70 @@
...
-  });
-});
+const VALID_STORAGE_OPTIONS_WITH_CUSTOM_ENC: StorageOptions[] = [
+  { environmentId: '', oauth: { token: '' }, getSecrets: () => '' },
+];
 
-describe('OAuth options validation', () => {
-  const codec = OAuthOptionsIO;
+describe('Storage options validation', () => {
+  [false].forEach((withCustomEnc) => {
```
Comment (artemgolozubov) [inline tests/unit/validation/storage-options-test.ts:36]:
```suggestion
  [false, true].forEach((withCustomEnc) => {
```

**Review by artemgolozubov — COMMENTED** [review]
Please take a look at the comments below

**Review by artemgolozubov — DISMISSED** [review]
LGTM

File: src/validation/user-input/storage-options.ts [hunk 4]
```diff
@@ -102,31 +38,59 @@ const CountriesCacheIO = new t.Type<CountriesCache>(
...
+  countriesCache?: CountriesCache;
+}>;
+
+const storageOptionsValidate = (hasCustomEncryptionConfigs: boolean) => (i: t.TypeOf<typeof StorageOptionsBasicIO>): Either<string, StorageOptionsValidated> => {
+  const encrypt = i.encrypt !== false;
+  const hashSearchKeys = i.hashSearchKeys !== false;
+  const getSecrets = encrypt ? i.getSecrets : undefined;
+
+  if (encrypt && getSecrets === undefined) {
+    return left('Provide callback function for secretData');
```
Comment (RommeroInCountry) [inline src/validation/user-input/storage-options.ts:78]: Provide a valid function returning 'SecretsData' object as 'getSecrets' StorageOptions attribute

**Review by RommeroInCountry — CHANGES_REQUESTED** [review]

File: src/validation/utils.ts [hunk 5]
```diff
@@ -147,15 +151,15 @@ const withAdditionalValidate = <A, B, BO, I>(type: t.Type<B, BO, I>, validate: (
   t.identity,
 );
 
-const validateArrayLength = <A>(arr: Array<A>): Either<string, Array<A>> => {
-  if (arr.length < 1) {
+const validateLength = <A extends { length: number }>(a: A): Either<string, A> => {
```
Comment (artemgolozubov) [inline src/validation/utils.ts:154]: Looks like `A` could also be a string here, so error message 'Array must be non empty' will look quite weird.

**Review by artemgolozubov — APPROVED** [review]
LGTM

File: src/validation/utils.ts (hunk 5 above)
Comment (igor-alekseev-dev) [reply src/validation/utils.ts:154]: It was an attempt to make `validateLength` more generic and accepts any type with `.length`

Comment (igor-alekseev-dev) [reply src/validation/utils.ts:154]: But it is still used only for array validation for now

**Review by RommeroInCountry — APPROVED** [review]

**Pull request merged**